Add Job Configuration to Mallus

G-Kemp101 opened the pull request:
Job configuration can now be specified by the user
If not specified an accept all config is loaded
A compare method is added to the job config which returns if the
sent config is bounded by their config
if so responds with accept: True else accept: False

G-Kemp101 committed:
Add Job Configuration to Mallus

Job configuration can now be specified by the user
If not specified an accept all config is loaded
A compare method is added to the job config which returns if the
sent config is bounded by their config
if so responds with accept: True else accept: False

G-Kemp101 committed:
MyPy corrections

G-Kemp101 committed:
added return statement to _process_job_config

G-Kemp101 committed:
Added newline to job_config

alexander-jackson commented on sybl/client/job_config.py:
```diff
@@ -0,0 +1,17 @@
+        timeout: int = job_config["timeout"]
+        prediction_type: str = job_config["prediction_type"].lower()
+
+        return timeout < self.max_timeout and prediction_type in self.prediction_types
```
Can we get some unit tests for this please

alexander-jackson commented on sybl/client/sybl_client.py:
```diff
@@ -225,21 +227,28 @@ def _process_job_config(self) -> bool:
+                self._send_message({"ConfigResponse": {"accept": False}})
+                self._state = State.HEARTBEAT
+                logger.info("REJECTING JOB")
+                return False
```
This section might be testable as well

FreddieBrown committed:
updated state machine for new behaviour (#13)

* updated state machine for new behaviour

* black reformat

* requested changes

* corrected CI issues

G-Kemp101 committed:
Add Tests for client

Add tests for sybl client and moves the authentication tests into a tests
directory

G-Kemp101 committed:
Add error handling found from tests

Added more Error handling and moved load_access_token
Also adds new method begin_state_machine so connect is easy to test
without the while True loop

G-Kemp101 committed:
Moved test_authenticate to testing folder

G-Kemp101 committed:
Pylint and Black fixes

G-Kemp101 committed:
Fix issue with moving load_access_token

G-Kemp101 committed:
Rename sybl fixutre sybl_instance

G-Kemp101 committed:
Merge pull request #17 from G-Kemp101/tests/config_tests

Tests/config tests

G-Kemp101 committed:
Disable pylint for redefined-outer-name

G-Kemp101 committed:
Merge pull request #18 from G-Kemp101/tests/config_tests

Disable pylint for redefined-outer-name

G-Kemp101 committed:
Job config return False on bad input

G-Kemp101 committed:
Add tests for JobConfig

G-Kemp101 committed:
Ignore function docstrings

G-Kemp101 committed:
Add tests for Job Config

Tests/config tests

G-Kemp101 committed:
Replace logging with lazy formatting

G-Kemp101 committed:
Merge branch 'feature/job_config' of github.com:G-Kemp101/mallus into feature/job_config

G-Kemp101 committed:
Pylint fixes including Docstrings

FreddieBrown committed:
Test/process job (#20)

* new tests

* fix test

* working tests

* remove debug statements

* Update sybl/client/sybl_client.py

Co-authored-by: Graham Kemp <44637230+G-Kemp101@users.noreply.github.com>

* Update tests/test_client.py

Co-authored-by: Graham Kemp <44637230+G-Kemp101@users.noreply.github.com>

* Update sybl/client/sybl_client.py

Co-authored-by: Graham Kemp <44637230+G-Kemp101@users.noreply.github.com>

* revert change

* moved function outside of Sybl

* shortens the code a little

* Fix/project test (#21)

* Hard code expected callback return value and Predictions

* Replace col indexing with insert and pop

* Improve pylint score

* Update tests/test_client.py

Co-authored-by: Alexander Jackson <alexanderjackson@protonmail.com>

Co-authored-by: Graham Kemp <44637230+G-Kemp101@users.noreply.github.com>
Co-authored-by: Alexander Jackson <alexanderjackson@protonmail.com>
